# Assignment
### Instructions
In the next series of challenges, we will predict employee salaries from different employee characteristics (or features). We ask employees working in four different fields to state their salary and some background information.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


In [2]:
# Reading in the data
salaries = pd.read_csv("salary.csv")
salaries.head()

salary  exprior  yearsworked  yearsrank  market  degree  otherqual  \
0  53000.0        0            0          0    1.17       1          0   
1  58000.0        1            0          0    1.24       1          0   
2  45500.0        0            0          0    1.21       1          0   
3  35782.0        0            2          1    0.99       1          0   
4  34731.0        0            2          2    0.91       1          0   

   position  male  Field  yearsabs  
0         1     1      3         0  
1         1     1      2         0  
2         1     1      3         2  
3         1     1      4         1  
4         1     1      4         1

In [3]:
salaries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 514 entries, 0 to 513
Data columns (total 11 columns):
salary         513 non-null float64
exprior        514 non-null int64
yearsworked    514 non-null int64
yearsrank      514 non-null int64
market         514 non-null float64
degree         514 non-null int64
otherqual      514 non-null int64
position       514 non-null int64
male           514 non-null int64
Field          514 non-null int64
yearsabs       514 non-null int64
dtypes: float64(2), int64(9)
memory usage: 44.3 KB


There's a total of 514 entries, 11 columns where 9 of them consist of interger values and the other two have float values.

In [4]:
salaries.isnull().sum()

salary         1
exprior        0
yearsworked    0
yearsrank      0
market         0
degree         0
otherqual      0
position       0
male           0
Field          0
yearsabs       0
dtype: int64

This information indicates that there's only one missing value under the column labeled "salary". All the other columns have no missing values.

In [5]:
salaries.describe()

salary     exprior  yearsworked   yearsrank      market  \
count    513.000000  514.000000   514.000000  514.000000  514.000000   
mean   50863.220098    2.926070    12.852140    7.052529    0.948521   
std    12685.132358    4.791397     9.444695    6.414771    0.149380   
min    29000.000000    0.000000     0.000000    0.000000    0.710000   
25%    40000.000000    0.000000     4.000000    2.000000    0.840000   
50%    50096.000000    0.500000    12.000000    5.000000    0.920000   
75%    60345.000000    4.000000    22.000000   12.000000    1.020000   
max    96156.000000   25.000000    41.000000   28.000000    1.330000   

           degree   otherqual    position        male       Field    yearsabs  
count  514.000000  514.000000  514.000000  514.000000  514.000000  514.000000  
mean     0.964981    0.044747    2.132296    0.750973    2.529183    6.982490  
std      0.184008    0.206950    0.820075    0.432871    1.127420   16.873156  
min      0.000000    0.000000    1.000000    0.000000    1.000000    0.000000  
25%      1.000000    0.000000    1.000000    1.000000    2.000000    0.000000  
50%      1.000000    0.000000    2.000000    1.000000    3.000000    1.000000  
75%      1.000000    0.000000    3.000000    1.000000    4.000000    2.000000  
max      1.000000    1.000000    3.000000    1.000000    4.000000  118.000000

- The lowest salary in the group is R29000.00 and the highest salary in the group is R96156.00
- The mean salary for the sample is R50863.22
- The standard error for the mean is 